peachjam/adapters/adapters.py

```diff
@@ -95,27 +95,35 @@ def update_document(self, url):
             "title": title,
             "created_at": document["created_at"],
             "updated_at": document["updated_at"],
-            "date": datetime.strptime(document["expression_date"], "%Y-%m-%d").date(),
             "content_html_is_akn": True,
             "source_url": document["publication_document"]["url"]
             if document["publication_document"]
             else None,
-            "jurisdiction": jurisdiction,
-            "locality": locality,
             "language": language,
             "toc_json": toc_json,
             "content_html": content_html,
+        }
+
+        frbr_uri_data = {
+            "locality": locality,
+            "jurisdiction": jurisdiction,
             "frbr_uri_subtype": frbr_uri.subtype,
             "frbr_uri_number": frbr_uri.number,
             "frbr_uri_doctype": frbr_uri.doctype,
+            "date": datetime.strptime(document["expression_date"], "%Y-%m-%d").date(),
+            "language": language,
         }
 
         if frbr_uri.actor:
-            field_data["frbr_uri_actor"] = frbr_uri.actor
+            frbr_uri_data["frbr_uri_actor"] = frbr_uri.actor
             field_data["author"] = Author.objects.update_or_create(code=frbr_uri.actor)[
                 0
             ]
 
+        doc = CoreDocument(**frbr_uri_data)
+        doc.work_frbr_uri = doc.generate_work_frbr_uri()
+        expression_frbr_uri = doc.generate_expression_frbr_uri()
```

Comment:
Add two asserts here: the generated work and expression FRBR URIs and the ones provided by indigo must match

Reply:
I have added the check below.